File: src/0011-Container-With-Most-Water/0011.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();

class Solution 
{
public:
    int maxArea(vector<int>& height) 
    {
        int l = 0;
        int r = height.size() - 1;
        int contain = 0;
        while (l < r)
        {
            contain = max(contain, (r - l)* min(height[l], height[r]));
            if (height[l] > height[r]) --r;
            else ++l;
        }
        return contain;
    }
};
int main()
{
    vector<int> height = {1, 1};
    cout << Solution().maxArea(height) << endl;
    return 0;
}

File: src/0011-Container-With-Most-Water/0011.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l = 0
        r = len(height) - 1
        contain = 0
        while l < r:
            contain = max(contain, (r - l) * min(height[l], height[r]))
            if height[l] > height[r]:
                r -= 1
            else:
                l += 1

        return contain

if __name__ == '__main__':
    height = [1,2,4,3]
    print(Solution().maxArea(height))
